paddle/framework/lod_tensor.cc

```diff
@@ -103,5 +111,139 @@ void LoDTensor::ShrinkInLevel(size_t level, size_t elem_begin,
   lod_ = new_lod;
 }
 
+std::string LoDTensor::SerializeToString() const {
```

Comment:
make `SerializeToString` an external function such as `SerializeToString(LoDTensor)`.

There may be more such serialization functions, such as `SerializeToString(OperatorBase)`, do not change the definition of the original class.

better not to insert methods that no relation with computation into `class LoDTensor`.

`LoDTensor` serves as a concept for computation, keep it clean.

and this function is too long, break it and keep the code clean.

if so much code is added and is no relation to the definition or operation of the concepts of `LoD` or `Tensor`, place it inside `namespace detail` or in another source file is better.

Reply:
After a talk with @Superjom face to face, my opinion on this question as below.
1. Currently, we only need to serialize the in-memory content into a byte-stream.  Namely, `SerializeToString(LoDTensor)`, `SerializeToString(Tensor)`. Operatorbase and other concepts all have their `Desc` in protobuf, we do not need any other class serializes implementation.

2. Implement `DeserilizeFromString` will return a Tensor filled with value, if we don't bind the serialize interface to the Tensor instance, we need another copy of the Tensor.

3. The `offset_` and `type` in Tensor is hidden. Need to figure them out.

Thanks for this comment!

Reply:
`DeserilizeFromString(LoDTensor*)` is no need to copy a Tensor, fill the data in-place seems possible.

Reply:
This concern is not replied and done before this PR merged ...